addons/binding/org.openhab.binding.cm11a/src/main/java/org/openhab/binding/cm11a/handler/Cm11aBridgeHandler.java

```diff
@@ -0,0 +1,229 @@
+    }
+
+    @Override
+    public void initialize() {
+        // Get serial port number from config
+        cm11aConfig = getThing().getConfiguration().as(Cm11aConfig.class);
+        logger.trace("********* cm11a initialize started *********");
+
+        // Verify the configuration is valid
+        if (!validateConfig(this.cm11aConfig)) {
+            return;
+        }
+
+        // Initialize the X10 interface
+        try {
+            x10Interface = new X10Interface(cm11aConfig.serialPort);
+            x10Interface.setDaemon(true);
+            x10Interface.start();
+            x10Interface.addReceivedDataListener(this);
+            logger.info("Initialized CM11A X10 interface on: {}", cm11aConfig.serialPort);
+        } catch (NoSuchPortException e) {
+            x10Interface = null;
+            logger.error("No such port exists on this machine: {}", cm11aConfig.serialPort);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "No such port exists on this machine: " + cm11aConfig.serialPort);
+            return;
+        }
+
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    @Override
+    public void dispose() {
+        // Close the serial port
+        if (x10Interface != null) {
+            x10Interface.disconnect();
+            x10Interface = null;
+        }
+    }
+
+    /**
+     * Validate that the configuration is valid
+     *
+     * @param cm11aConfig2
+     * @return
+     */
+    private boolean validateConfig(Cm11aConfig config) {
+        if (config == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "cm11a configuration missing");
+            return false;
+        }
+
+        String port = config.serialPort;
+        if (port == null || port.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "cm11a serialPort not specified");
+            return false;
+        }
+
+        return true;
+    }
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see org.openhab.binding.cm11a.handler.ReceivedDataListener#receivedX10Data(org.openhab.binding.cm11a.internal.
+     * X10ReceivedData)
+     */
+    @Override
+    public void receivedX10Data(X10ReceivedData rd) {
+        logger.debug("Cm11aReceivedDataManager received the following data: {}", rd);
+
+        List<Thing> things = bridge.getThings();
+
+        // This block goes through the Things attached to this bridge and find the HouseUnitCode that matches what came
+        // from the serial port. Then it looks at the channels in that thing and looks for a channel that ends with
+        // "switchstatus" or "lightlevel"
+        // which is the one that should be updated.
+        synchronized (rd) {
+            for (Thing thing : things) {
+                String houseUnitCode = (String) thing.getConfiguration().get("HouseUnitCode");
+                for (String huc : rd.getAddr()) {
```

Comment:
Maybe give this a more clear name like `messageHouseUnitCode`, `messageAddress`, `targetAddress`